addons/website_sale_delivery/models/sale_order.py

```diff
@@ -75,19 +76,20 @@ def _get_delivery_methods(self):
         # Following loop is done to avoid displaying delivery methods who are not available for this order
         # This can surely be done in a more efficient way, but at the moment, it mimics the way it's
         # done in delivery_set method of sale.py, from delivery module
-        carrier_ids = DeliveryCarrier.sudo().search(
-            [('website_published', '=', True)]).ids
-        for carrier_id in carrier_ids:
-            carrier = DeliveryCarrier.browse(carrier_id)
+        user = request.env.user
+        if user.has_group('base.group_website_publisher') and user.has_group('sales_team.group_sale_manager'):
+            carriers = DeliveryCarrier.search([])
+        else:
+            carriers = DeliveryCarrier.sudo().search([('website_published', '=', True)])
+        for carrier in carriers:
             try:
-                _logger.debug("Checking availability of carrier #%s" % carrier_id)
-                available = carrier.with_context(order_id=self.id).read(fields=['available'])[0]['available']
-                if available:
+                _logger.debug("Checking availability of carrier #%s" % carrier.id)
+                if carrier.with_context(order_id=self.id).available:
```

Comment:
That is not efficient, because changing the context will make a new query for each carrier. You should directly searchf or carriers with the right context above.